C#/forSpbu/FindPair/GameCore.cs

```diff
@@ -0,0 +1,100 @@
+namespace FindPare;
+
+public class GameCore
+{
+    enum State
+    {
+        firstButtonPressed,
+        secondButtonPressed
+    }
+    public GameCore(int fieldSize)
+    {
+        _field = new int[fieldSize][];
+        for (int i = 0; i < fieldSize; i++)
+        {
+            _field[i] = new int[fieldSize];
+        }
+        _isOpened = new bool[fieldSize][];
+        for (int i = 0; i < fieldSize; i++)
+        {
+            _isOpened[i] = new bool[fieldSize];
+        }
+        _dimensionSize = fieldSize;
+        _state = State.secondButtonPressed;
+        RandomiseField();
+    }
+
+    private void RandomiseField()
+    {
+        var cellValues = new List<int>(Enumerable.Range(0, _dimensionSize * _dimensionSize / 2));
```

Comment:
```suggestion
        var cellValues = new List<int>(Enumerable.Range(0, _dimensionSize * _dimensionSize / 2 - 1));
```
Более того, _dimensionSize * _dimensionSize / 2 - 1 стоило вынести в отдельную переменную, несколько раз это выражение в методе используется